addons/website_sale/models/website.py

```diff
@@ -675,3 +683,9 @@ def _get_checkout_steps(self, current_step=None):
         if current_step:
             return next(step for step in steps if current_step in step[0])[1]
         return steps
+
+    def has_shop_access(self):
+        """ Return whether the current user is allowed to access eCommerce-related content. """
+        return not (self.env.user._is_public() and self.ecommerce_access == 'logged')
```

Comment:
Should be `logged_in`.

Reply:
Indeed, that's the reason it was broken.